include/evmc/evmc.hpp

```diff
@@ -408,6 +410,12 @@ class VM
         return result{m_instance->execute(m_instance, &host, ctx, rev, &msg, code, code_size)};
     }
 
+    inline result execute(Host& host,
```

Comment:
The rest of the methods here are inline in the class, why define this outside of class?

Reply:
Should it have some doxygen comment?

Reply:
Because the implementation requires Host and Host is defined after VM. Maybe reordering the definitions can help, but that will require another PR to have diff readable.